app/events/services/service_deleted_event.rb

```diff
@@ -12,7 +12,7 @@ def self.create(service)
         provider_id: provider.id
       }
     }
-    data[:provider] = provider if provider.persisted?
+    data[:provider] = provider if provider.persisted? && !provider.scheduled_for_deletion?
```

Comment:
I prefer you rescue in the activejob instead
If the object is not found just ignore it

Reply:
@hallelujah I can rescue but I do not think that we should ignore in this case... This ServiceDeletedEvent may be used for more things (or not, I do not know), but for deleting in Backend in particular, we do not need the provider to be still persisted (or even to have its data in the event at all, except for the id, which we already do), and we want it to be removed from Backend anyway.